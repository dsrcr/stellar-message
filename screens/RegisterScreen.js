import { FontAwesome5 } from "@expo/vector-icons";
import { Button, Image, Input } from "@rneui/base";
import { useFormik } from "formik";
import { Text, View, Pressable } from "react-native";
import tw from "twrnc";
import Yup from "yup";
export default function RegisterScreen({ navigation }) {
  return (
    <View style={tw`items-center flex-1 justify-center p-8`}>
      <>
        <Input
          id="email"
          style={tw`pt-4`}
          autoComplete="email"
          placeholder="E-mail Address"
        />
        <Input id="password" secureTextEntry placeholder="Password" />
        <Input id="password" secureTextEntry placeholder="Confirm password" />
      </>

      <Button title={"Register"} containerStyle={tw`m-4 p-4 w-80`} />

      <View style={tw`flex-row`}>
        <Text>Already have an account? </Text>
        <Pressable onPress={() => navigation.navigate("Login")}>
          <Text>Log in</Text>
        </Pressable>
      </View>
    </View>
  );
}
