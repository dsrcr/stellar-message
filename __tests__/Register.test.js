import React from 'react';
import renderer from 'react-test-renderer';

import RegisterScreen from '../screens/RegisterScreen';

describe('<RegisterScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RegisterScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
