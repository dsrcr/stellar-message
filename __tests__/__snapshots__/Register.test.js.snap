// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "paddingBottom": 32,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 32,
    }
  }
>
  <TextInput
    autoFocus={true}
    id="email"
    inputMode="email"
    keyboardType="email-address"
    name="email"
    onChangeText={[Function]}
    placeholder="email-address"
    placeholderTextColor="gray"
    style={
      {
        "borderColor": "#d1d5db",
        "borderRadius": 12,
        "borderWidth": 2,
        "fontSize": 16,
        "lineHeight": 24,
        "marginTop": 32,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
    textContentType="emailAddress"
    value=""
  />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#d1d5db",
        "borderRadius": 12,
        "borderWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 32,
        "paddingLeft": 16,
        "paddingRight": 20,
        "width": "100%",
      }
    }
  >
    <TextInput
      id="password"
      inputMode="text"
      name="password"
      onChangeText={[Function]}
      placeholder="password"
      placeholderTextColor="gray"
      secureTextEntry={true}
      style={
        {
          "borderRadius": 12,
          "fontSize": 16,
          "lineHeight": 24,
          "paddingBottom": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
      textContentType="password"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text />
    </View>
  </View>
  <TextInput
    id="confirmPassword"
    inputMode="text"
    name="confirmPassword"
    onChangeText={[Function]}
    placeholder="confirm-password"
    placeholderTextColor="gray"
    secureTextEntry={true}
    style={
      {
        "borderColor": "#d1d5db",
        "borderRadius": 12,
        "borderWidth": 2,
        "fontSize": 16,
        "lineHeight": 24,
        "marginTop": 32,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
    textContentType="password"
    value=""
  />
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        },
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 12,
            "borderWidth": 0,
            "flexDirection": "row",
            "fontSize": 18,
            "justifyContent": "center",
            "lineHeight": 28,
            "padding": 12,
            "paddingHorizontal": 14,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          register
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "lineHeight": 24,
        }
      }
    >
      already-have
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "color": "#2089DC",
            "fontSize": "1rem",
          }
        }
      >
        login
      </Text>
    </View>
  </View>
</View>
`;
